syntax = "proto3";

package proto.schema;

// The Greeter service offers personalized greeting services.
service Greeter {
    // Sends a personalized greeting to the client based on the provided name and language preference.
    rpc SayHello (GreetingRequest) returns (GreetingResponse) {}

    // Retrieves a summary of all greetings sent to a specific user.
    rpc GetGreetingHistory (GreetingHistoryRequest) returns (GreetingHistoryResponse) {}
}

// The HelloRequest message contains the input data for sending a greeting.
message GreetingRequest {
    string salutation = 1;
    string name = 2;
}

// The HelloResponse message contains the server's response to a greeting request.
message GreetingResponse {
    string message = 1;
    string timestamp = 2;
}


// The GreetingHistoryRequest message is used to request a user's greeting history.
message GreetingHistoryRequest {
    string name = 1;
}

// The GreetingHistoryResponse message contains the response to a greeting history request.
message GreetingHistoryResponse {
    repeated string messages = 1;
    repeated string timestamp = 2;
}
